src/core/js/models/lockingModel.js

```diff
@@ -0,0 +1,143 @@
+define([
+	'backbone'
+], function() {
+
+	var set = Backbone.Model.prototype.set;
+
+	_.extend(Backbone.Model.prototype, {
+
+		set: function(attrName, attrVal, options) {
+			var stopProcessing = !this.lockedAttributes || typeof attrName === "object" || typeof attrVal !== "boolean" || !this.isLocking(attrName);
+			if (stopProcessing) return set.apply(this, arguments);
+
+			var isSettingValueForSpecificPlugin = options && options.pluginName;
+			if (!isSettingValueForSpecificPlugin) {
+				console.error("Must supply a pluginName to change a locked attribute");
+				options.pluginName = "compatibility";
```

Comment:
This is going to cause an error if set is called and options is undefined. Only a problem I guess if you have a combination of old and new plugins setting a locked value. e.g latest assessmentResults and trickle v2.0. This combination breaks the build. Would this be considered a breaking change?